### DISCUSSION TITLE
I need help setting up Twitter SignUp

### DISCUSSION THREAD

**Discussion by JumboDaniel** [nextauthjs/next-auth#7739] [Help]
### Question 💬

When I click on the sign-in button on my Frontend it redirects me to the Twitter SignIn page, after signing in it returns `Something went wrong. You weren’t able to give access to the App. Go back and try logging in again.`. 
I searched for the error online, I saw a GitHub issue where it said that the problem is from the redirect URI. I used the recommended:
[http:/127.0.0.1:3001/api/auth/twitter],
[http://localhost:3000/api/auth/callback/twitter] 
but it still gives the same error.

Thanks for your assistance.

### How to reproduce ☕️

/app/api/auth/[...nextauth]/route.ts
`
import NextAuth from "next-auth";
import TwitterProvider from "next-auth/providers/twitter";
import type { NextAuthOptions } from "next-auth";

  const options: NextAuthOptions = {
    providers: [
      TwitterProvider({
        clientId: process.env.TWITTER_CLIENT_ID,
        clientSecret: process.env.TWITTER_CLIENT_SECRET,
        version: "2.0",

      }),
    ],
  };

const handler = NextAuth(options)
export { handler as GET, handler as POST };
`

**page.tsx**
`"use client";

import { Metadata } from "next"
import Link from "next/link"

import { cn } from "@/lib/utils"
import { buttonVariants } from "@/components/ui/button"
// import { Icons } from "@/components/icons"
// import { UserAuthForm } from "@/components/user-auth-form"
import { signIn, signOut, useSession } from 'next-auth/react';

export const metadata: Metadata = {
  title: "Login",
  description: "Login to your account",
}

export default function LoginPage() {

  return (
    <div className="container flex h-screen w-screen flex-col items-center justify-center">
      <Link
        href="/"
        className={cn(
          buttonVariants({ variant: "ghost" }),
          "absolute left-4 top-4 md:left-8 md:top-8"
        )}
      >
        <>
          {/* <Icons.chevronLeft className="mr-2 h-4 w-4" /> */}
          Back
        </>
      </Link>      <div className="mx-auto flex w-full flex-col justify-center space-y-6 sm:w-[350px]">
        <div className="flex flex-col space-y-2 text-center">
          {/* <Icons.logo className="mx-auto h-6 w-6" /> */}

          <p className="text-sm text-muted-foreground">
            Enter your email to sign in to your account
          </p>
        </div>
        <button  onClick={() => signIn('twitter',{ callbackUrl: 'http://127.0.0.1:3001/api/auth/callback/twitter'})} className="cursor-pointer">
          Sign In
        </button>
        {/* <UserAuthForm /> */}
        <p className="px-8 text-center text-sm text-muted-foreground">
          <Link
            href="/register"
            className="hover:text-brand underline underline-offset-4"
          >
            Don&apos;t have an account? Sign Up
          </Link>
        </p>
      </div>
    </div>
  )
}`

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by hohoaisan** [comment]
have nextAuth ever confirmed to work with /app directory?
Might be duplicated from this https://github.com/nextauthjs/next-auth/discussions/4034

**Reply by JumboDaniel** [reply]
Yes, it works with /app directory. You need to use catch-all routes in the API setup.
https://next-auth.js.org/getting-started/example
https://next-auth.js.org/configuration/nextjs
Check out the docs

**Comment by zkSoju** [comment]
Were you able to find a solution for this?

**Reply by JumboDaniel** [reply]
Yes, Turns out that my callback URI was not set up properly.
NEXTAUTH_URL= http://localhost:3000
this is the proper configuration

**Comment by jokester** [comment]
The symptom in OP can be caused by different reasons.

I had the same error message in my app and found that my cause was incorrectly copied OAuth2 `clientId`.

In my case the cause is identified by this response (in Twitter's error page):

```
GET https://twitter.com/i/api/2/oauth2/authorize?.....
=> http 400
{
    "error": "invalid_client",
    "error_description": "Value passed for the client id was invalid."
}
```

Hope it helps
